imap_processing/mag/l1d/mag_l1d.py

```diff
@@ -45,16 +45,18 @@ def mag_l1d(
     input_mago_burst = None
     for dataset in science_data:
         source = dataset.attrs.get("Logical_source", "")
-        if "norm-magi" in source:
-            input_magi_norm = dataset
-        elif "norm-mago" in source:
-            input_mago_norm = dataset
-        elif "burst-magi" in source:
-            input_magi_burst = dataset
-        elif "burst-mago" in source:
-            input_mago_burst = dataset
-        else:
-            raise ValueError(f"Input data has invalid logical source {source}")
+        ins_mode = source.split("_")[-1]
+        match ins_mode:
```

Comment:
instrument? Be explicit if it is just a few extra characters in this one spot.
```suggestion
        instrument_mode = source.split("_")[-1]
        match instrument_mode:
```